PWGJE/Tasks/PhotonIsolationQA.cxx

```diff
@@ -0,0 +1,364 @@
+/// \author Berend van Beuzekom <b.d.vanbeuzekom@students.uu.nl>
+/// \since 30-05-2024
+///
+/// This task is designed to select EMCal clusters originating from prompt photons.
+/// Cluster selection is performed using Pt_Iso (density of particles around the cluster minus UE density) and cluster shape.
+/// The task can also be run over Monte Carlo data to obtain a correction factor for the purity measurement using the ABCD method.
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+using myGlobTracks = o2::soa::Join<o2::aod::FullTracks, o2::aod::TrackSelection>;
+using collisionEvSelIt = o2::soa::Join<o2::aod::Collisions, o2::aod::EvSels>;
+using mcCells = o2::soa::Join<aod::Calos, aod::McCaloLabels_001>;
+using filteredMCCells = o2::soa::Filtered<mcCells>;
+using MCClusters = o2::soa::Join<o2::aod::EMCALMCClusters, o2::aod::EMCALClusters>;
+
+struct PhotonIsolationQA {
+  HistogramRegistry Data_Info{"Data_Info", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+  HistogramRegistry MC_Info{"MC_Info", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+
+  using o2HistType = HistType;
+  using o2Axis = AxisSpec;
+
+  Configurable<float> maxPosZ{"maxPosZ", 10.0f, "maximum z position of collision in cm"};
+  Configurable<std::string> eventSelections{"eventSelections", "sel8", "choose event selection"};
+  Configurable<std::string> trackSelections{"trackSelections", "globalTracks", "set track selections"};
+  Configurable<int> mClusterDefinition{"clusterDefinition", 0, "cluster definition to be selected, e.g. 10=kV3Default, 0 = kV1Default"};
+  Configurable<float> minTime{"minTime", -30., "Minimum cluster time for time cut"};
+  Configurable<float> maxTime{"maxTime", +35., "Maximum cluster time for time cut"};
+  Configurable<float> minClusterEnergy{"minClusterEnergy", 0.7f, "Minimal cluster energy"};
+  Configurable<int> minNCells{"minNCelss", 2, "Minimal amount of cells per cluster"};
+  Configurable<int> maxNLM{"maxNLM", 2, "Maximal amount of local maxima per cluster"};
+  Configurable<bool> ExoticContribution{"ExoticContribution", false, "Exotic cluster in the data"};
+  Configurable<float> TrackMatchingRadius{"TrackMatchingRadius", 0.05, "maximum radius between track and cluster in eta phi plane"};
+  Configurable<float> minDPhi{"minDPhi", 0.01, "Minimum dPhi between track and cluster"};
+
+  Filter PosZFilter = nabs(aod::collision::posZ) < maxPosZ;
+  Filter clusterDefinitionSelection = (o2::aod::emcalcluster::definition == mClusterDefinition) && (o2::aod::emcalcluster::time >= minTime) && (o2::aod::emcalcluster::time <= maxTime) && (o2::aod::emcalcluster::energy > minClusterEnergy) && (o2::aod::emcalcluster::nCells >= minNCells) && (o2::aod::emcalcluster::nlm <= maxNLM) && (o2::aod::emcalcluster::isExotic == ExoticContribution);
+  Filter emccellfilter = aod::calo::caloType == 1; // mc emcal cell
+
+  using selectedCollisions = soa::Filtered<collisionEvSelIt>;
```

Comment:
what is collisionEvSelIt?

Reply:
it is defined in line 58: using collisionEvSelIt = o2::soa::Join<o2::aod::Collisions, o2::aod::EvSels>;